core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/WorkflowResource.scala

```diff
@@ -90,13 +108,16 @@ class WorkflowResource {
   def persistWorkflow(@Session session: HttpSession, workflow: Workflow): Response = {
     UserResource.getUser(session) match {
       case Some(user) =>
-        if (workflowOfUserExists(workflow.getWid, user.getUid)) {
+        if (WorkflowAccessResource.hasWriteAccess(workflow.getWid, user.getUid)) {
```

Comment:
actually, a second thought. this endpoint's behavior needs more discussion. We should change it to the following conditions:
1. is owner but has no access -> insert proper access level, then persist the workflow as wish.
2. is owner but has only read access -> can this happen? we can ignore this case, treat it as case 1.
3. is owner and has write access -> persist the workflow as wish.
4. is not owner and has no access -> reject.
5. is not owner but has only read access -> reject.
6. is not owner but has write access -> persist the workflow as wish.

Reply:
I modified the code snippet in a slightly differnt logic. 
When the endpoint is requested, we are either creating a new workflow or modifying an existing workflow. In the "New Workflow" case, there will be no record for ownership or record for access since ownership and access level are defined at the same time and /persist is the only way to modify these two properties (in my understanding). In the "Modifying an existing workflow" case, a user must be granted certain privilege before being allowed to access the workflow, otherwise they will be blocked. Therefore:

1. is owner but has no access -> not applicable/cannot happen
2. is not owner and has no accessRecord -> not applicable/cannot happen
3. is owner but has only read access -> we can ignore this case, treat it as case 4.
4. is owner and has write access -> persist the workflow as wish.
5. is not owner and has no accessRecord -> insert access level and ownership, and persist workflow.
6. is not owner but has only read access -> reject
7. is not owner but has write access -> persist the workflow as wish.
8. is not owner and has no access -> not applicable/cannot happen

I have modified the code as above (I did not include the cases that are not applicable, will add those in next commit for safety precautions)